### PR TITLE
Add option to export reconciliation data in CSV

### PR DESCRIPTION
**Pull request by shahbaz-mehar**

### CODE REVIEW

**Review by BushraAsif — COMMENTED** [review]
changes reviewed

**Comment by emicha** [comment]
image
This will be very hard to parse, or even to use by a regular user when they download the CSV.

Why not splitting id, type and status into its own columns?

**Comment by shahbaz-mehar** [comment]
> image
> 
> This will be very hard to parse, or even to use by a regular user when they download the CSV.
> Why not splitting id, type and status into its own columns?

I have updated the functionality. Now we are saving the reconciliation data in our custom table and provided an option to export it to CSV.

File: classes/core/AltapayReconciliation.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+			$output    = '';
+			$file_name = 'reconciliation_data.csv';
+
+			header( 'Content-Type: text/csv; charset=utf-8' );
+			header( 'Content-Disposition: attachment; filename=' . $file_name );
+			header( 'Pragma: no-cache' );
+			header( 'Expires: 0' );
+
+			$reconciliation_data = $wpdb->get_results( "SELECT * FROM {$wpdb->prefix}altapayReconciliationIdentifiers", ARRAY_A );
```
Comment (emicha) [inline classes/core/AltapayReconciliation.php:106]: It seems that you are selecting everything from the table, which will start failing later on due to the amount of data you will have.

Why not using the filters that you see in the screen, or the orders that are being displayed to be exported?

Comment (shahbaz-mehar) [reply classes/core/AltapayReconciliation.php]: @emicha I have updated the query to select the orders displayed on the page.

Comment (emicha) [reply classes/core/AltapayReconciliation.php]: @shahbaz-mehar I still see "SELECT * FROM table"

**Review by emicha — APPROVED** [review]
